cluster/terraform_kubernetes/grafana.tf

```diff
@@ -0,0 +1,119 @@
+data "azurerm_key_vault" "key_vault_grafana" {
+  name                = var.cluster_kv
+  resource_group_name = var.resource_group_name
+}
+
+data "azurerm_key_vault_secret" "grafana_admin_user" {
+  name         = "GRAFANA-ADMIN-USER"
+  key_vault_id = data.azurerm_key_vault.key_vault_grafana.id
+}
+
+data "azurerm_key_vault_secret" "grafana_admin_password" {
+  name         = "GRAFANA-ADMIN-PASSWORD"
+  key_vault_id = data.azurerm_key_vault.key_vault_grafana.id
+}
+
+
+resource "kubernetes_config_map" "grafana_datasources" {
+  metadata {
+    name      = "grafana-datasources"
+    namespace = "monitoring"
+  }
+
+  data = {
+    "datasources.yaml" = file("${path.module}/config/prometheus/datasources.yaml")
+  }
+}
+
+
+resource "kubernetes_deployment" "grafana_deployment" {
+  metadata {
+    name      = "grafana"
+    namespace = "monitoring"
+    labels = {
+      app = "grafana"
+    }
+  }
+
+  spec {
+    replicas = 1
+
+    selector {
+      match_labels = {
+        app = "grafana"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "grafana"
+        }
+      }
+
+      spec {
+        container {
+          name  = "grafana"
+          image = "grafana/grafana:latest"
```

Comment:
can you add grafana_version as a variable. Set it to default to the current latest version (10.3.3?).